cli/flox/src/commands/init.rs

```diff
@@ -262,6 +275,528 @@ impl InitHook for Requirements {
+    /// available.
+    fn get_package_json_version(path: &Path, flox: &Flox) -> Result<PackageJSONVersion> {
+        let package_json = path.join("package.json");
+        if !package_json.exists() {
+            return Ok(PackageJSONVersion::None);
+        }
+        let package_json_contents = fs::read_to_string(package_json)?;
+        match serde_json::from_str::<serde_json::Value>(&package_json_contents) {
+            // Treat a package.json that can't be parsed as JSON the same as it not existing
+            Err(_) => Ok(PackageJSONVersion::None),
+            Ok(package_json_json) => match package_json_json["engines"]["node"].as_str() {
+                Some(version) => match Self::check_for_node_version(version, flox)? {
+                    None => Ok(PackageJSONVersion::Unavailable),
+                    Some(result) => Ok(PackageJSONVersion::Some(Box::new(result))),
+                },
+                None => Ok(PackageJSONVersion::Unspecified),
+            },
+        }
+    }
+
+    fn check_for_node_version(version: &str, flox: &Flox) -> Result<Option<SearchResult>> {
+        let query = Query {
+            name: None,
+            pname: Some("nodejs".to_string()),
+            version: None,
+            semver: Some(version.to_string()),
+            r#match: None,
+            match_name: None,
+            match_name_or_rel_path: None,
+            limit: Some(1),
+            deduplicate: false,
+        };
+        let params = SearchParams {
+            manifest: None,
+            global_manifest: PathOrJson::Path(global_manifest_path(flox)),
+            lockfile: PathOrJson::Path(LockedManifest::ensure_global_lockfile(flox)?),
+            query,
+        };
+
+        let (mut results, _) = do_search(&params)?;
+
+        if results.results.is_empty() {
+            return Ok(None);
+        }
+        Ok(Some(results.results.swap_remove(0)))
+    }
+
+    fn get_default_node(flox: &Flox) -> Result<SearchResult> {
+        let query = Query::new("nodejs", Features::parse()?.search_strategy, Some(1), false)?;
+        let params = SearchParams {
+            manifest: None,
+            global_manifest: PathOrJson::Path(global_manifest_path(flox)),
+            lockfile: PathOrJson::Path(global_manifest_lockfile_path(flox)),
+            query,
+        };
+
+        let (mut results, _) = do_search(&params)?;
+
+        if results.results.is_empty() {
+            Err(anyhow!("Flox couldn't find any versions of nodejs"))?
+        }
+        Ok(results.results.swap_remove(0))
+    }
+
+    /// Return whether to skip the nodejs hook entirely, install a requested
+    /// version of nodejs, or offer to install the Flox default version of
+    /// nodejs.
+    ///
+    /// This is decided based on whether .nvmrc and package.json are present,
+    /// and whether Flox can provide versions they request.
+    fn get_action(&self) -> NodeAction {
```

Comment:
@ghudgins @jennymahmoudi @brendaneamon @tanjadev

I made a bunch of decisions about what we should say and do for all of these cases, but I'm not sure which of them make sense and if wording for prompts is good.

Reply:
`declaration` should be renamed to `manifest` in these strings

Reply:
otherwise it looks good to us (reviewed with @jennymahmoudi )

Reply:
if package.json's engines are not compatible with flox we don't suggest anything--even if there is an .nvmrc.

if there is a package.json engine it should win over the .nvmrc

Reply:
engines in package.json can also have "yarn": "^1.22.19"

Reply:
npm may also be specified in the package.json's engines. this may be its own ticket

Reply:
> `declaration` should be renamed to `manifest` in these strings

I also changed revisit to change because revisit the manifest sounded weird to me

Reply:
> if package.json's engines are not compatible with flox we don't suggest anything--even if there is an .nvmrc.
> 
> if there is a package.json engine it should win over the .nvmrc

Done